services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/MetersValidationSingleSwFlowSpec.groovy

```diff
@@ -0,0 +1,406 @@
+        Wrappers.wait(WAIT_OFFSET) {
+            def switchValidateInfoAfterDelete = northbound.switchValidate(sw.dpId)
+            ruleSectionsAreEmpty(switchValidateInfoAfterDelete)
+            meterSectionsAreEmpty(switchValidateInfoAfterDelete)
+        }
+
+        where:
+        switchType   | switches               | flagsOnSwitch
+        "Centec"     | getCentecSwitches()    | ["KBPS", "BURST", "STATS"]
+        "non-Centec" | getNonCentecSwitches() | ["KBPS", "BURST", "STATS"]
+    }
+
+    @Unroll
+    def "Switch validation is able to detect meter info into the 'excess' section on a #switchType switch"() {
+        assumeTrue("Unable to find required switches in topology", switches as boolean)
+
+        setup: "Select a #switchType switch and retrieve default meters"
+        def sw = switches.first()
+
+        when: "Create a flow"
+        def flow = flowHelper.addFlow(flowHelper.singleSwitchFlow(sw))
+        def metersIds = createdMeterIds(sw.dpId)
+        Long burstSize = switchHelper.getExpectedBurst(sw.dpId, flow.maximumBandwidth)
+
+        and: "Update meterId for created flow directly via db"
+        def newMeterId = 100
+        database.updateFlowMeterId(flow.id, newMeterId)
+
+        then: "Meters info are moved into the 'excess' section"
+        def switchValidateInfo = northbound.switchValidate(sw.dpId)
+        switchValidateInfo.meters.excess.collect { it.meterId }.containsAll(metersIds)
+        switchValidateInfo.meters.excess.each {
+            assert it.rate == flow.maximumBandwidth
+            assert flagsOnSwitch.containsAll(it.flags)
+            burstSizeIsCorrect(burstSize, it.burstSize)
+        }
+
+        and: "Meters info is also stored in the 'missing' section"
+        def createdCookies = сookiesWithMeter(sw.dpId)
+        switchValidateInfo.meters.missing.collect { it.cookie }.containsAll(createdCookies)
+        switchValidateInfo.meters.missing.each {
+            assert it.rate == flow.maximumBandwidth
+            assert it.flowId == flow.id
+            assert it.meterId == newMeterId
+            assert flagsOnSwitch.containsAll(it.flags)
+            burstSizeIsCorrect(burstSize, it.burstSize)
+        }
+
+        and: "Rules are still exist in the 'proper' section"
+        ruleSectionsAreEmpty(switchValidateInfo, ["missing", "excess"])
+
+        when: "Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+
+        and: "Delete created meters on a switch"
+        metersIds.each { northbound.deleteMeter(sw.dpId, it) }
+
+        then: "Check that switch validate info is also deleted"
+        Wrappers.wait(WAIT_OFFSET) {
+            def switchValidateInfoAfterDelete = northbound.switchValidate(sw.dpId)
+            ruleSectionsAreEmpty(switchValidateInfoAfterDelete)
+            meterSectionsAreEmpty(switchValidateInfoAfterDelete)
+        }
+
+        where:
+        switchType   | switches               | flagsOnSwitch
```

Comment:
Why `flagsOnSwitch` is present in iterations variables? It does not change between iterations and should be removed from here